src/http-libwebsockets.c

```diff
@@ -361,6 +386,135 @@ static void connection_wake(qd_connection_t *qd_conn)
     }
 }
 
+static void handle_stats_results(void *context, qdr_global_stats_t *results)
+{
+    stats_t* stats = (stats_t*) context;
+    qd_http_server_t *hs = stats->server;
+    if (hs) {
+        work_t w = { W_HANDLE_STATS, stats->wsi };
+        work_push(hs, w);
```

Comment:
It should be noted that this handler is being called on the core-thread.  Since work_push can block on a condition, the core thread is at risk for being blocked by the HTTP processing.  It might be better for the core-thread to schedule this callback as a general-work item so it runs on an I/O thread.

Reply:
Agreed. What's the call for doing that can you remind me?

Reply:
Since the metrics consists only of counters, it's possible that they could be accessed synchronously by the I/O thread handling the metric request without using a lock.  There is the possibility that any given response might be inconsistent (i.e. being written at the same time as read), but that might be acceptable in this case.

Reply:
I believe the nature of the interaction means this will never happen. Each request uses a separate instance of the counters. Only the core thread updates those counters and they are only read by the http thread once the core thread has made the callback.

Reply:
Sorry, I meant to not even use an action for the core thread.  Just go read the current values of the counters from the HTTP thread, ignoring any glitches that occur as a result of concurrency.

If you use the general-work method, the call is qdr_post_general_work.  I think there's a void* you can use to point to the metric object and you can add fields to qdr_general_work_t if needed.  You of course need to write the I/O-thread-side work function that calls the metric callback.